Lib/posixpath.py

```diff
@@ -538,19 +538,20 @@ def relpath(path, start=None):
         raise
 
 
-# Return the longest common sub-path of the sequence of paths given as input.
+# Return the longest common sub-path of the iterable of paths given as input.
 # The paths are not normalized before comparing them (this is the
 # responsibility of the caller). Any trailing separator is stripped from the
 # returned path.
 
 def commonpath(paths):
-    """Given a sequence of path names, returns the longest common sub-path."""
+    """Given an iterable of path names, returns the longest common sub-path."""
 
-    paths = tuple(map(os.fspath, paths))
+    paths = tuple(paths)
 
     if not paths:
-        raise ValueError('commonpath() arg is an empty sequence')
+        raise ValueError('commonpath() arg is an empty iterable')
 
+    _, roots, paths = zip(*map(splitroot, paths))
```

Comment:
```suggestion
    _, roots, paths = zip(*map(splitroot, paths))
    roots.sort()
```

Reply:
Oh sorry, the `roots.sort()` line needs to be inside the `try:` block below to handle mixed str/bytes.